internal/app/machined/pkg/runtime/v1alpha1/bootloader/bootloader.go

```diff
@@ -67,14 +74,89 @@ func NewAuto() Bootloader {
 	return grub.NewConfig()
 }
 
-// New returns a new bootloader based on the secureboot flag.
-func New(secureboot bool, talosVersion string) Bootloader {
-	if secureboot {
-		return sdboot.New()
+// New returns a new bootloader based on the secureboot flag and architecture.
+func New(bootloader, talosVersion, arch string) (Bootloader, error) {
+	if arch == "arm64" {
```

Comment:
This should still honor the bootloader option? Probably instead of dual it should do sdboot, but it should do grub still if explicitly configured?

Reply:
not sure I get it? this honors the option, the profile fills the default to be `dual-boot` but users can change to any other and this still works

Reply:
I mean here on arm64 it always uses sd-boot, while it should honor setting, and use grub/sd-boot?